### PR TITLE
Added Mystery Number

### LINKED ISSUE #1148: Check if a number is a Mystery Number in C++, Java, Python, Go

### PR DESCRIPTION
**Pull request by MastersAbh**
Fixes #1148  

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Added Implementation for Mystery Number in C, C++, Java, Python, Golang

#### Languages Used:
- C
- C++
- Java
- Python
- Golang

#### Files Added:
- Mystery_Number.py
- Mystery_Number.c
- Mystery_Number.cpp
- Mystery_Number.java
- Mystery_Number.go

### CODE REVIEW

**Comment by somya-kapoor** [comment]
I approve of the PR, please have a look. @jainaman224

File: Mystery_Number/Mystery_Number.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+int solve(int num) {
+    for (int i = 1; i <= num / 2; i++) {
+        if (rev(i) == num - i) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+int main(){
```
Comment (jainaman224) [inline Mystery_Number/Mystery_Number.c]: Space before `{` will be great.

File: Mystery_Number/Mystery_Number.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+int solve(int num) {
+    for (int i = 1; i <= num / 2; i++) {
+        if (rev(i) == num - i) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+int main(){
```
Comment (jainaman224) [inline Mystery_Number/Mystery_Number.cpp]: Space before `{` will be great.

File: Mystery_Number/Mystery_Number.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/*
+    MYSTERY NUMBER
+
+    A Mystery Number is a number that can be expressed as a sum of two numbers,
+    where those two numbers are reverse of each other.
+*/
+
+public class Mystery{
```
Comment (jainaman224) [inline Mystery_Number/Mystery_Number.java]: Space before `{` will be great.

File: Mystery_Number/Mystery_Number.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    A Mystery Number is a number that can be expressed as a sum of two numbers,
+    where those two numbers are reverse of each other.
+'''
+
+'''  Approach : We iterate over from 1 to n/2, to get all the pairs of number, that
+                can sum up to the given no. n. and check if the numbers in the pair 
+                are reverse of each other.
+'''
+def rev(num) :
```
Comment (jainaman224) [inline Mystery_Number/Mystery_Number.py]: I am not sure but I think, there is convention that no space before `:` in python.
Even in hash etc.

**Comment by MastersAbh** [comment]
Done. Please review @jainaman224

**Pull request merged**